lib/matrix_check_sum.rb

```diff
@@ -3,8 +3,32 @@
 # of numbers in row i is the same as the sum of numbers in column i for i = 0 to row.length-1
 # If this is the case, return true. Otherwise, return false.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2), where n is the number of rows/columns
+# Space complexity: O(n), where n is the number of rows/columns ?
 def matrix_check_sum(matrix)
-  raise NotImplementedError
+  num_of_rows = matrix.length
+
+  sum_rows = []
+  num_of_rows.times do |i|
+    sum = 0
+    num_of_rows.times do |j|
+      sum += matrix[i][j]
+    end
+    sum_rows << sum
```

Comment:
In this loop you're finding the sum of each row.  Could you also find the sum of the corresponding column at the same time?